### PR TITLE
Issue 6483 and 6482: Flaky tests.

### LINKED ISSUE #6483: Flaky test GarbageCollectorTests.testMaxAttempts()
**Describe the bug**
Found test failing sporadically in Git Actions:
```
io.pravega.segmentstore.storage.chunklayer.GarbageCollectorTests > testMaxAttempts FAILED
    java.lang.AssertionError: expected:<1> but was:<0>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at io.pravega.segmentstore.storage.chunklayer.GarbageCollectorTests.assertQueueEquals(GarbageCollectorTests.java:1325)
        at io.pravega.segmentstore.storage.chunklayer.GarbageCollectorTests.testMaxAttempts(GarbageCollectorTests.java:1194)
```

**To Reproduce**
Sporadically reproducible running `./gradlew build`

**Screenshots**
n/a

**Additional information**
n/a

### PR DESCRIPTION
**Pull request by sachin-j-joshi**
Signed-off-by: Sachin Joshi <sachin.joshi@emc.com>

**Change log description**  
LTS - Flaky tests - fix GarbageCollectorTests.testMaxAttempts() and ignore ChunkMetadataStoreTests.testEvictionFromBuffer().

**Purpose of the change**  
Temp fix for #6482 
Fixes #6483

**What the code does**  

- Fixes GarbageCollectorTests.testMaxAttempts()
- Temporarily disables ChunkMetadataStoreTests.testEvictionFromBuffer()

**How to verify it**  
Tests should pass.

### CODE REVIEW

File: segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/GarbageCollectorTests.java [hunk 1]
```diff
@@ -1157,7 +1157,7 @@ public void testMaxAttempts() throws Exception {
         Function<Duration, CompletableFuture<Void>> noDelay = d -> CompletableFuture.completedFuture(null);
         val testTaskQueue = new InMemoryTaskQueueManager();
 
-        chunkStorage.setReadOnly(chunkStorage.openWrite("deletedChunk").get(), true);
+        chunkStorage.setReadOnly(chunkStorage.openWrite("deletedChunk").get(), true).join();
```
Comment (sachin-j-joshi) [inline segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/GarbageCollectorTests.java:1160]: Forgot to wait :)

**Review by abhinb — APPROVED** [review]

**Comment by derekm** [comment]
@sachin-j-joshi -- Does `testEvictionFromBuffer` fail all the time or just some of the time? If it only fails some of the time, should we keep the test and make it easier for non-committers to re-run tests?

I have a github bot that can re-run PR Actions workflows based on PR comments. I haven't set it up in production for Pravega yet, but I could expedite this if it helps us keep tests active that only sometimes fail.

Otherwise, it feels like ignoring the test could lead to the test never being fixed.

FWIW, you can test the re-run bot here: https://github.com/derekm/pravega-github-bot/pull/1

**Comment by sachin-j-joshi** [comment]
Test fails intermittently because of timing issues and needs to be fixed or rewritten. The code is covered indirectly by other tests. This test just tests an extreme scenario - and fails sometimes. 

However to reduce false failures I'm temporarily  disabling the tests with clear goal of either replacing it or fixing it shortly.

**Review by RaulGracia — COMMENTED** [review]
Will merge this one, but please make sure to give a closure on `testEvictionFromBuffer()`.

**Review by RaulGracia — APPROVED** [review]

**Pull request merged**